### PR TITLE
fix:[CORE-1546] fix for scroll api in qualys asset data importer job

### PR DESCRIPTION
**Pull request by raviteja-mandala**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List
any dependencies that are required for this change.

Fixes # (issue)

Qualys asset data importer is throwing data alert because of issue in scroll api. Resolved issue with necessary changes.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My commit message/PR follows the contribution guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java [hunk 1]
```diff
@@ -344,10 +349,15 @@ public static Map<String, String> fetchEniMacInfo() {
...
-        } while (scrollId != null);
+
+        while (scrollId != null) {
+            endPoint = "/_search/scroll";
+            payLoad = "{\n" +
+                    "  \"scroll\" : \"2m\", \n" +
+                    "  \"scroll_id\" : \"%s\" \n" +
+                    "}";
+            scrollId = fetchDataAndScrollId(endPoint, data, String.format(payLoad, scrollId));
+        }
```
Comment (plyubich) [inline jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java:360]: This looks like duplicate code, please extract to a separate function

**Review by plyubich — CHANGES_REQUESTED** [review]
Please extract duplicate code into a separate function

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java (hunk 1 above)
Comment (raviteja-mandala) [reply jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java:361]: done.

**Review by santhosh-challa — APPROVED** [review]
LGTM

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/Main.java [hunk 2]
```diff
@@ -70,9 +70,15 @@ public static Map<String, Object> execute(Map<String, String> params) throws Nam
             break;
         }
         ArrayList errors= (ArrayList) errorInfo.get("errors");
+
         if(!errors.isEmpty()){
             //Below logger message is used by datadog to create notification in slack
             log.error("Error occurred in atleast one collector for jobId : {}",jobHint);
+            errors.stream().forEach(error -> {
+                Map<String, Object> err = (Map<String, Object>) error;
+                log.error("error in main - {}", err.get("error"));
```
Comment (plyubich) [inline jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/Main.java:79]: Here for each error you are creating two log statements. Let's just do one log.error with the full error exception for each error?

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java [hunk 3]
```diff
@@ -32,10 +32,11 @@ public static Map<String,Object> formErrorCode(List<Map<String,String>> errorLis
             for(Map<String, String> errorDetail :errorList) {
                 Map<String,Object> error = new HashMap<>();
                 error.put(ERROR, errorDetail.get(ERROR));
-                
+                log.error("error detail - {}", errorDetail.get(ERROR));
```
Comment (plyubich) [inline jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java:35]: Same here. To many log statements. Let's combine into a single log statement for each error

**Review by plyubich — CHANGES_REQUESTED** [review]

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java (hunk 3 above)
Comment (raviteja-mandala) [reply jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java:35]: done.

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/Main.java (hunk 2 above)
Comment (raviteja-mandala) [reply jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/Main.java:79]: done.

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java [hunk 4]
```diff
@@ -344,16 +340,27 @@ public static Map<String, String> fetchEniMacInfo() {
...
         String scrollId = fetchDataAndScrollId(endPoint, data, payLoad);
-        do {
-            endPoint = SCROLL_URI + scrollId;
-            scrollId = fetchDataAndScrollId(endPoint, data, null);
-        } while (scrollId != null);
+
+        extractDataUsingScrollId(scrollId, data);
 
         return data.stream().collect(Collectors.toMap(info -> info.get("_resourceid").toLowerCase(),
                 info -> info.get("macaddress").toLowerCase()));
```
Comment (plyubich) [inline jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/jobs/Util.java:348]: Extra new line

File: jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java [hunk 5]
```diff
@@ -18,6 +16,8 @@ private ErrorManageUtil() {
         
     }
 
+    private static Logger log = LoggerFactory.getLogger(ErrorManageUtil.class);
+
     public static Map<String,Object> formErrorCode(List<Map<String,String>> errorList) {
         Map<String,Object> errorCode = new HashMap<>();
         //to do write error file
```
Comment (plyubich) [inline jobs/pacman-qualys-enricher/src/main/java/com/tmobile/cso/pacman/qualys/util/ErrorManageUtil.java:23]: Change to do -> TODO:

**Review by plyubich — APPROVED** [review]
Approved, suggestion - minor updates - see comments

**Pull request merged**